autogen/agentchat/contrib/swarm_agent.py

```
@@ -122,6 +122,11 @@ def swarm_transition(last_speaker: SwarmAgent, groupchat: GroupChat):
         if tool_execution._next_agent is not None:
             next_agent = tool_execution._next_agent
             tool_execution._next_agent = None
+
+            # Check for string, access agent from group chat.
+            if isinstance(next_agent, str):
+                next_agent = groupchat.agent_by_name(name=next_agent)
```

If the agent name doesn't exist then it will return None, we should raise an error here if that's the case. So `next_agent` will be None.

Alternatively, you can check in the `swarm_agent_names` list to see if an agent name exists at this point.

At the moment, if the name doesn't exist the chat just terminates without any error message.